src/jrd/ExtEngineManager.cpp

```diff
@@ -703,6 +717,13 @@ void* ExtEngineManager::ExternalContextImpl::setInfo(int code, void* value)
 	return oldValue;
 }
 
+void ExtEngineManager::ExternalContextImpl::checkExternalAttachment()
+{
+	if (!externalAttachment)
+	{
```

Comment:
If check for presence of external attachment is anyway needed - why not move creation of it here from constructor? I agree that overhead due to unused externalAttachment is not too big, but a lot of UDR functions (one who perform supposed activity based on passed parameters, i.e. have UDF-like functionality) do not need it at all.

Reply:
Will do, thanks